csrf_test.go

```diff
@@ -272,6 +272,48 @@ func TestBadReferer(t *testing.T) {
 	}
 }
 
+// TestTrustedReferer checks that HTTPS requests with a Referer that does not
+// match the request URL correctly but is a trusted origin pass CSRF validation.
+func TestTrustedReferer(t *testing.T) {
+	s := http.NewServeMux()
+
+	p := Protect(testKey, TrustedOrigins([]string{"golang.org"}))(s)
```

Comment:
Can you provide a table-driven test with a bunch of possible referer values? e.g.

```
var testTable := []struct{
    trustedOrigin []string
    shouldPass bool
}{
    {[]string{"golang.org"}, true},
    {[]string{"golang.org", "api.example.com", true},
    {[]string{"https://example.com/foo"}, true},
    // etc
}
```

e.g. provide a list of valid, invalid referrers - which will both describe support (valid Referers), self-document the matching, and prevent regressions.